heidrun/ct_dc.rb

```diff
@@ -0,0 +1,95 @@
+
+# If more than one dc:contributor property exists then it drops the last one
+# If one or zero exist return nil
+contributor_top = lambda do |r|
+  contrib = r['dc:contributor']
+  count = r['dc:contributor'].count
+  (count > 1) ? contrib.first_value(count - 1) : contrib.last_value(0)
+end
+
+Krikri::Mapper.define(:ct_dc,
+                      :parser => Krikri::OaiDcParser,
+                      :parser_args => '//rdf:RDF/rdf:Description') do
+  provider :class => DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/connecticut'
+    label 'Connecticut Digital Library'
+  end
+
+  dataProvider :class => DPLA::MAP::Agent do
+    providedLabel record.field('dc:contributor').last_value
+  end
+
+  isShownAt :class => DPLA::MAP::WebResource do
+    uri record.field('dc:identifier')
+  end
+
+  preview :class => DPLA::MAP::WebResource do
+    uri record.field('dc:source')
+  end
+
+  originalRecord :class => DPLA::MAP::WebResource do
+    uri record_uri
+  end
+
+  sourceResource :class => DPLA::MAP::SourceResource do
+    collection  :class => DPLA::MAP::Collection,
+                :each => record.field('xmlns:header', 'xmlns:setSpec'),
+                :as => :coll do
+      title coll
+    end
+
+    contributor :class => DPLA::MAP::Agent,
+                :each => record.map( &contributor_top ).flatten,
```

Comment:
That seems reasonable, though it sends us back to the original problem.

It turns out we don't have access to `#compact` here, so we won't be able to squash out the `nil` value that way. I'm throwing together a patch to introduce that method.

In the meanwhile, we could consider `reject { |v| v.nil? }`.